component/config.go

```diff
@@ -155,14 +157,18 @@ func MustNewType(strType string) Type {
 // collecting metrics, traces and logs, this can expand in the future.
 type DataType = Type
```

Comment:
I am keeping this for now, but we can revisit once we tackle #9429

Reply:
It is an alias to Type because we use `component.ID` (which is not generic) for the `pipeline.ID`. Because of that we made this equal with Type.